Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ReviewRequestController.java

```diff
@@ -0,0 +1,35 @@
+package edu.asu.conceptpower.web;
+
+import java.security.Principal;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import edu.asu.conceptpower.app.core.impl.RequestsManager;
+import edu.asu.conceptpower.core.ReviewStatus;
+import edu.asu.conceptpower.core.RequestJsonResponse;
+import edu.asu.conceptpower.core.ReviewRequest;
+
+@Controller
+public class ReviewRequestController {
+ 
+    @Autowired
+    private RequestsManager requestsMgr;
+           
+    @RequestMapping(value = "/auth/request/add", method = RequestMethod.POST )
+    public @ResponseBody RequestJsonResponse addNewReviewRequest( @ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
+  
+        reviewRequest.setStatus(ReviewStatus.OPENED);
+        reviewRequest.setRequester(principal.getName());
+        
+        RequestJsonResponse res = new RequestJsonResponse();
```

Comment:
what do you need this class for? It has only one field, which is the review object. Why don't you just return the review object?